### ISSUE TITLE
Pydantic string fields incorrectly reject enum members in lax mode

### ISSUE THREAD

**Issue by carljm** [astral-sh/ty#4259]
### Summary

Pydantic accepts enum members as inputs to `str` fields in its default lax validation mode, converting them with `str(member.value)`. However, ty reports `invalid-argument-type` for these valid constructor arguments.

```python
from enum import Enum

from pydantic import BaseModel, ConfigDict

class Color(Enum):
    RED = "red"

class Number(Enum):
    SEVEN = 7

class Model(BaseModel):
    value: str

class StrictModel(BaseModel):
    model_config = ConfigDict(strict=True)
    value: str

Model(value="red")         # Accepted by both Pydantic and ty.
Model(value=Color.RED)     # Pydantic returns "red"; ty incorrectly reports invalid-argument-type.
Model(value=Number.SEVEN)  # Pydantic returns "7"; ty incorrectly reports invalid-argument-type.

StrictModel(value=Color.RED)  # Correctly rejected: strict mode does not coerce ordinary enums.
```

### Expected behavior

ty should accept enum members as inputs to `str` fields when Pydantic validation is lax, regardless of the type of the enum member’s value. Strict models and strict fields should continue rejecting ordinary enum members.

### Cause

ty currently models lax string inputs as:

```python
type LaxStr = bytearray | bytes | str
```

This omits `Enum`. Pydantic’s documented conversion rules suggest:

```python
type LaxStr = bytearray | bytes | str | Enum
```

See [Pydantic’s string validation documentation](https://docs.pydantic.dev/latest/api/standard_library_types/#strings).

It seems like our conversion support was based on a conversion table that [may not be up to date](https://github.com/astral-sh/ty/issues/2403#issuecomment-4913853689), but adding support for converting enums seems reasonable given that it is explicitly documented as supported.

**Comment by charliermarsh** [comment]
I can take this while David is out.

**Comment by charliermarsh** [comment]
Also going to audit that table against the code more completely.
